src/Microsoft.ML.ImageAnalytics/Imager.cs

```diff
@@ -0,0 +1,420 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using SkiaSharp;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.ML.Data
+{
+    /// <summary>
+    /// Provide interfaces for imaging operations.
+    /// </summary>
+    public class Imager : IDisposable
+    {
+        private SKBitmap _image;
+
+        /// <summary>
+        /// Create a new imager instance from a stream.
+        /// </summary>
+        /// <param name="imageStream">The stream to create the image from.</param>
+        public Imager(Stream imageStream)
+        {
+            if (imageStream is null)
+            {
+                throw new ArgumentNullException(nameof(imageStream));
+            }
+
+            _image = SKBitmap.Decode(imageStream);
+            if (_image is null)
+            {
+                throw new ArgumentException($"Invalid input stream contents", nameof(imageStream));
+            }
+
+            EnsureSupportedPixelFormat();
+        }
+
+        /// <summary>
+        /// Create a new imager instance from image file.
+        /// </summary>
+        /// <param name="imagePath">The image file path to create the image from.</param>
+        public Imager(string imagePath)
```

Comment:
Similar to my questions about the stream API - when is this file read and does a handle remain open?

Reply:
I have replied that in the comment https://github.com/dotnet/machinelearning/pull/6363#discussion_r995178890